server/channels/api4/ip_filtering.go

```diff
@@ -83,34 +81,7 @@ func applyIPFilters(c *Context, w http.ResponseWriter, r *http.Request) {
 
 	auditRec.Success()
 
-	cloudWorkspaceOwnerEmailAddress := ""
-	if c.App.License().IsCloud() {
-		portalUserCustomer, cErr := c.App.Cloud().GetCloudCustomer(c.AppContext.Session().UserId)
-		if cErr != nil {
-			c.Logger.Error("Failed to get portal user customer", mlog.Err(cErr))
-		}
-		if cErr == nil && portalUserCustomer != nil {
-			cloudWorkspaceOwnerEmailAddress = portalUserCustomer.Email
-		}
-	}
-
-	go func() {
-		initiatingUser, err := c.App.Srv().Store().User().GetProfileByIds(context.Background(), []string{c.AppContext.Session().UserId}, nil, true)
-		if err != nil {
-			c.Logger.Error("Failed to get initiating user", mlog.Err(err))
-		}
-
-		users, err := c.App.Srv().Store().User().GetSystemAdminProfiles()
-		if err != nil {
-			c.Logger.Error("Failed to get system admins", mlog.Err(err))
-		}
-
-		for _, user := range users {
-			if err = c.App.Srv().EmailService.SendIPFiltersChangedEmail(user.Email, initiatingUser[0], *c.App.Config().ServiceSettings.SiteURL, *c.App.Config().CloudSettings.CWSURL, user.Locale, cloudWorkspaceOwnerEmailAddress == user.Email); err != nil {
-				c.Logger.Error("Error while sending IP filters changed email", mlog.Err(err))
-			}
-		}
-	}()
+	go c.App.SendIPFiltersChangedEmail(c.AppContext, c.AppContext.Session().UserId)
```

Comment:
super nit: Should we make the decision to do this asynchronous something "inside the app layer" vs. a choice here?

Reply:
I'd think the opposite actually since it lets the caller decide that this should be done asynchronously

Reply:
I do agree as a Go-ism, but I'm wondering if the app layer should be the one deciding to spawn goroutines vs. the API layer? For example, we sometimes use `c.App.Srv().Go(...)` to do this this, though not consistently.